[Doc] Replace fields of xxx by xxx field (#1599)
taichi-dev/taichi: docs/meta.rst

There are several reasons why ``ti.static`` for loops should be used.

 - Loop unrolling for performance.

>  - Loop over vector/matrix elements. Indices into Taichi matrices must be a compile-time constant. Indexing into taichi fields can be run-time variables. For example, if ``x`` is a 1-D vector field which has 3-D vector on each entry, accessed as ``x[field_index][matrix_index]``. The first index can be variable, yet the second must be a constant.

Reviewer: Clear a bit confusion.

```suggestion
 - Loop over vector/matrix elements. Indices into Taichi matrices must be a compile-time constant. Indexing into taichi fields can be run-time variables. For example, if ``**x**`` is a 3-D vector field of the shape of 1 dimension, accessed as ``**x**[field_index][matrix_index]``. The first index can be variable, yet the second must be a constant.
```

Reviewer 2: ```suggestion
 - Loop over vector/matrix elements. Indices into Taichi matrices must be a compile-time constant. Indexing into taichi fields can be run-time variables. For example, if you want to access a vector field ``x``, accessed as ``x[field_index][vector_component_index]``. The first index can be variable, yet the second must be a constant.
```
